projects/dsp-ui/src/lib/core/session.service.ts

```diff
@@ -0,0 +1,215 @@
+        @Inject(DspApiConnectionToken) private dspApiConnection: KnoraApiConnection,
+        @Inject(DspApiConfigToken) private dspApiConfig: KnoraApiConfig
+    ) { }
+
+    /**
+     * get session information from localstorage
+     */
+    getSession(): Session {
+        if (localStorage.getItem('session') !== null) {
+            return JSON.parse(localStorage.getItem('session'));
+        }
+
+        return undefined;
+    }
+
+    /**
+     * set session by using the json web token (jwt) and the user object;
+     * it will be used in the login process
+     *
+     * @param jwt
+     * @param username
+     */
+    setSession(jwt: string, identifier: string, identifierType: 'email' | 'username') {
+
+        let session: Session;
+
+        if (jwt) {
+            this.updateDspApiConnection(jwt);
+        }
+
+        // get user information
+        this.dspApiConnection.admin.usersEndpoint.getUser(identifierType, identifier).subscribe(
+            (response: ApiResponseData<UserResponse>) => {
+                let sysAdmin = false;
+                const projectAdmin: string[] = [];
+
+                // get permission information: a) is user sysadmin? b) get list of project iri's where user is project admin
+                const groupsPerProjectKeys: string[] = Object.keys(response.body.user.permissions.groupsPerProject);
+
+                for (const key of groupsPerProjectKeys) {
+                    if (key === Constants.SystemProjectIRI) {
+                        sysAdmin = response.body.user.permissions.groupsPerProject[key].indexOf(Constants.SystemAdminGroupIRI) > -1;
+                    }
+
+                    if (response.body.user.permissions.groupsPerProject[key].indexOf(Constants.ProjectAdminGroupIRI) > -1) {
+                        projectAdmin.push(key);
+                    }
+                }
+
+                // store session information in browser's localstorage
+                // TODO: jwt will be removed, when we have a better cookie solution (DSP-261)
+                session = {
+                    id: this.setTimestamp(),
+                    user: {
+                        name: response.body.user.username,
+                        jwt: jwt,
+                        lang: response.body.user.lang,
+                        sysAdmin: sysAdmin,
+                        projectAdmin: projectAdmin
+                    }
+                };
+
+                // update localStorage
+                localStorage.setItem('session', JSON.stringify(session));
+            },
+            (error: ApiResponseError) => {
+                localStorage.removeItem('session');
+                console.error(error);
+            }
+        );
+    }
+
+    /**
+     * Validate intern session and check knora api credentials if necessary.
+     * If a json web token exists, it doesn't mean that the knora api credentials are still valid.
+     *
+     * @returns boolean
+     */
+    isSessionValid(): boolean {
```

Comment:
I think since this method involves a call to the Knora API, it can only return an `Observable<boolean>`. In the branches where no connection to Knora is necessary, you can return of(true) or `of(false)`.